### PR TITLE
Add ShadowSpringBootJpa Example

### LINKED ISSUE #15813: [OpenForce 2022]  Add `shadow-spring-boot-jpa-example` for `shadow-example`
Hi community,

This issue is for [OpenForce 2022](https://github.com/apache/shardingsphere/discussions/12683#discussioncomment-2139441)

### Background
What is feature Shadow?
https://shardingsphere.apache.org/document/current/en/features/shadow/

[shadow-example](https://github.com/apache/shardingsphere/tree/master/examples/shardingsphere-jdbc-example/single-feature-example/shadow-example) is used to demonstrate how to integrate ShardingSphere-JDBC and enable `shadow`.

Now, it has three sub-modules:

```
<module>shadow-raw-jdbc-example</module>
<module>shadow-spring-boot-mybatis-example</module>
<module>shadow-spring-namespace-mybatis-example</module>
```

### Aim

Add `shadow-spring-boot-jpa-example` for `shadow-example`

### Basic Qualifications

- Java
- Maven

### Example FYI

- [Configuration Item Explanation](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-jdbc/spring-boot-starter/rules/shadow/) 

- [sharding-spring-boot-jpa-example](https://github.com/apache/shardingsphere/tree/master/examples/shardingsphere-jdbc-example/single-feature-example/sharding-example/sharding-spring-boot-jpa-example)

### Notice
- You need to clone the ShardingSphere project from [here](https://github.com/apache/shardingsphere) first
- In the ShardingSphere directory, `examples` is a separate maven project, you should open `examples` instead of `ShardingSphere` in the IDE.
- Contributor Guide: https://shardingsphere.apache.org/community/en/contribute/contributor/
- Code of Conduct: https://shardingsphere.apache.org/community/en/contribute/code-conduct/

### PR DESCRIPTION
**Pull request by everly-gif**
Fixes #15813 .

I have added the spring boot jpa example to shadow-example.

Let me know if it looks good.

Thanks.

### CODE REVIEW

File: examples/shardingsphere-jdbc-example/single-feature-example/shadow-example/shadow-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/boot/jpa/ShadowSpringBootJpaExample.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.boot.autoconfigure.transaction.jta.JtaAutoConfiguration;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+
+import java.sql.SQLException;
+
+@ComponentScan("org.apache.shardingsphere.example.core.jpa")
+@EntityScan(basePackages = "org.apache.shardingsphere.example.core.jpa.entity")
+@SpringBootApplication(exclude = JtaAutoConfiguration.class)
+
```
Comment (yx9o) [inline examples/shardingsphere-jdbc-example/single-feature-example/shadow-example/shadow-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/boot/jpa/ShadowSpringBootJpaExample.java:34]: Please delete blank lines.

Comment (everly-gif) [reply examples/shardingsphere-jdbc-example/single-feature-example/shadow-example/shadow-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/boot/jpa/ShadowSpringBootJpaExample.java]: removed

Comment (everly-gif) [reply examples/shardingsphere-jdbc-example/single-feature-example/shadow-example/shadow-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/boot/jpa/ShadowSpringBootJpaExample.java]: @yx90 do you want to remove other spaces in the file as well? I have structured it the same way as in the example file given ShardingSpringBootJpaExample.java , but I noticed there was one extra blank line in my example and I have removed it. Can you tell me if you want me to remove the other blank lines as well?

**Review by RaigorJiang — CHANGES_REQUESTED** [review]
Hi @everly-gif 
Due to the addition of sub-module, the parent pom file also needs to be updated.
https://github.com/everly-gif/shardingsphere/blob/dev-15813/examples/shardingsphere-jdbc-example/single-feature-example/shadow-example/pom.xml

**Comment by RaigorJiang** [comment]
@yx9o  Thanks for your careful review!

**Comment by everly-gif** [comment]
@RaigorJiang  I have updated it, is there anymore suggested changes?

File: examples/shardingsphere-jdbc-example/single-feature-example/shadow-example/shadow-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/boot/jpa/ShadowSpringBootJpaExample.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.example.shadow.spring.boot.jpa;
+
+import org.apache.shardingsphere.example.core.api.ExampleExecuteTemplate;
+import org.apache.shardingsphere.example.core.api.service.ExampleService;
+import org.mybatis.spring.annotation.MapperScan;
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/shadow-example/shadow-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/boot/jpa/ShadowSpringBootJpaExample.java:22]: Please remove unused import  `org.mybatis.spring.annotation.MapperScan`.

Comment (everly-gif) [reply examples/shardingsphere-jdbc-example/single-feature-example/shadow-example/shadow-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/boot/jpa/ShadowSpringBootJpaExample.java]: Made the changes

**Comment by RaigorJiang** [comment]
Hi @everly-gif 
Thanks for your effort, it's good enough. It can be merged after CI passed.

**Comment by RaigorJiang** [comment]
@everly-gif 
Please fix the failed CI.

**Comment by everly-gif** [comment]
Hi @RaigorJiang I fixed the issue, I had missed an import statement. Could you re-trigger the build?

**Comment by everly-gif** [comment]
Hi @RaigorJiang  all tests have passed except for jdk 8 on macos latest. I'm not sure why could you help me resolve it?

**Comment by RaigorJiang** [comment]
> Hi @RaigorJiang all tests have passed except for jdk 8 on macos latest. I'm not sure why could you help me resolve it?

Failure is not related to this PR, don't worry.  I think it can be merged now.

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**